server.go

```diff
@@ -461,3 +465,25 @@ func sendConvertedFile(w http.ResponseWriter, filePath string) error {
 
 	return nil
 }
+
+func isMn(r rune) bool {
+	return unicode.Is(unicode.Mn, r) // Mn: nonspacing marks
+}
+
+func filenameToAscii7(s string) string {
+	// Remove most diacritics
+	t := transform.Chain(norm.NFD, transform.RemoveFunc(isMn), norm.NFC)
+	noDiacr, _, _ := transform.String(t, s)
+
+	// Convert the rest of non-ASCII7 to hex representation
+	var sb strings.Builder
```

Comment:
Would prefer using `strings.Map` here and using `unicode.MaxASCII` to make it clearer what the magic number represents.